core/whistle_services-1.0.0/src/bookkeepers/wh_bookkeeper_braintree.erl

```diff
@@ -471,42 +469,61 @@ handle_quick_sale_response(BtTransaction) ->
 %%
 %% @end
 %%--------------------------------------------------------------------
--spec already_charged(ne_binary(), integer() | ne_binary()) -> boolean().
--spec already_charged(ne_binary(), ne_binary(), wh_json:objects()) -> boolean().
-already_charged(BillingId, Code) ->
+-spec already_charged(ne_binary() | integer() , integer() | wh_json:objects()) -> boolean().
+already_charged(BillingId, Code) when is_integer(Code) ->
     lager:debug("checking if ~s has been charge for transaction of type ~p today", [BillingId, Code]),
     BtTransactions = braintree_transaction:find_by_customer(BillingId),
-    Transactions = [braintree_transaction:record_to_json(BtTransaction) || BtTransaction <- BtTransactions],
-    already_charged(BillingId, wh_util:to_binary(Code), Transactions).
+    Transactions = [braintree_transaction:record_to_json(BtTransaction)
+                    || BtTransaction <- BtTransactions
+                   ],
+    already_charged(wh_util:to_binary(Code), Transactions);
 
-already_charged(_, _, []) ->
+already_charged(_, []) ->
     lager:debug("no transactions found matching code or made today"),
     'false';
-already_charged(BillingId, Code, [Transaction|Transactions]) ->
-    case wh_json:get_value(<<"purchase_order">>, Transaction) of
-        Code ->
-            already_charged_transaction(BillingId, Code, Transaction, Transactions);
-        _TrCode -> already_charged(BillingId, Code, Transactions)
+already_charged(Code, [Transaction|Transactions]) ->
+    case
+        already_charged_transaction(
+          Code
+          ,wh_json:get_value(<<"status">>, Transaction)
+          ,wh_json:get_value(<<"purchase_order">>, Transaction)
+          ,Transaction
+         )
+    of
+        'true' -> 'true';
+        'false' ->
+            already_charged(Code, Transactions)
     end.
 
--spec already_charged_transaction(ne_binary(), ne_binary(), wh_json:object(), wh_json:objects()) ->
-                                         boolean().
-already_charged_transaction(BillingId, Code, Transaction, Transactions) ->
+%%--------------------------------------------------------------------
+%% @private
+%% @doc
+%%
+%% @end
+%%--------------------------------------------------------------------
+-spec already_charged_transaction(integer(), ne_binary(), integer(), wh_json:object()) -> boolean().
+already_charged_transaction(_ , ?BT_TRANS_VOIDED, _, Transaction) ->
+    Id = wh_json:get_value(<<"id">>, Transaction),
```

Comment:
When only using a Var for logging, prepend it with '_' (`_Id` in this case, and in the next clause).